src/cmdstan/command.hpp

```diff
@@ -295,29 +222,79 @@ int command(int argc, const char *argv[]) {
     write_parallel_info(sample_writers[i]);
     write_opencl_device(sample_writers[i]);
     write_compile_info(sample_writers[i], model_compile_info);
-    write_stan(diagnostic_writers[i]);
-    write_model(diagnostic_writers[i], model.model_name());
-    parser.print(diagnostic_writers[i]);
-  }
-
-  int refresh = get_arg_val<int_argument>(parser, "output", "refresh");
-
-  // arg "init" can be filename or non-negative number (init_radius)
-  std::string init = get_arg_val<string_argument>(parser, "init");
-  double init_radius = 2.0;
-  try {
-    init_radius = boost::lexical_cast<double>(init);
-    init = "";
-  } catch (const boost::bad_lexical_cast &e) {
+    if (!diagnostic_csv_writers.empty()) {
+      write_stan(diagnostic_csv_writers[i]);
+      write_model(diagnostic_csv_writers[i], model.model_name());
+      parser.print(diagnostic_csv_writers[i]);
+    }
   }
-  std::vector<std::shared_ptr<stan::io::var_context>> init_contexts
-      = get_vec_var_context(init, num_chains);
-  int return_code = return_codes::NOT_OK;
 
   //////////////////////////////////////////////////
   //            Invoke Services                   //
   //////////////////////////////////////////////////
-  if (user_method->arg("generate_quantities")) {
+  int return_code = return_codes::NOT_OK;
+  std::stringstream msg;
+
+  if (user_method->arg("pathfinder")) {
+    // ---- pathfinder start ---- //
+    auto pathfinder_arg = parser.arg("method")->arg("pathfinder");
+    int history_size
+        = get_arg_val<int_argument>(*pathfinder_arg, "history_size");
+    double init_alpha
+        = get_arg_val<real_argument>(*pathfinder_arg, "init_alpha");
+    double tol_obj = get_arg_val<real_argument>(*pathfinder_arg, "tol_obj");
+    double tol_rel_obj
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_obj");
+    double tol_grad = get_arg_val<real_argument>(*pathfinder_arg, "tol_grad");
+    double tol_rel_grad
+        = get_arg_val<real_argument>(*pathfinder_arg, "tol_rel_grad");
+    double tol_param = get_arg_val<real_argument>(*pathfinder_arg, "tol_param");
+    int max_lbfgs_iters
+        = get_arg_val<int_argument>(*pathfinder_arg, "max_lbfgs_iters");
+    bool save_iterations
+        = get_arg_val<bool_argument>(*pathfinder_arg, "save_iterations");
```

Comment:
This flag is still not used anywhere in the following code, which has hardcoded true/false values for the same-named service function argument

Reply:
This needs to be tested

Reply:
right.  testing this, and there's  definitely a problem here.  seems to be always saving iters, no matter what the setting is.

Reply:
As currently written it should always save iterations for multi-path pathfinder and never save them for single path. This is assuming it is all correct in the services layer, of course

Reply:
this is the problem:   "save_iterations" means dump lbfgs information to diagnostic_writer.  setting "save_iterations=1" (true), only makes sense if the output sub-arg "diagnostic_file" is specified.  we shouldn't expose "save_iterations" at all; instead, set the bool value "save_iterations" to whether or not arg "diagnostic_file" is specified.

Reply:
removed "save_iterations" as user-facing argument.  feature tested via tests for diagnostic_file.